diff --git a/CMakeLists.txt b/CMakeLists.txt
index cb47e51900..ae5717dae7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,11 +136,6 @@ else()
 endif()
 message(STATUS "CMAKE_INSTALL_INCLUDEDIR: ${CMAKE_INSTALL_INCLUDEDIR}")
 
-# On macOS we want BUNDLEs (.app) to go to an 'Applications/' directory instead of a 'bin/' directory
-if(APPLE)
-  set(OM_MACOS_INSTALL_BUNDLEDIR "Applications/")
-endif()
-
 ## Set the adjusted installation lib directory as an rpath for all installed binaries.
 ## Assumes binaries end up in either 'bin' or 'lib' AND also assumes that 'bin' and 'lib' are sibling directories.
 if(APPLE)
diff --git a/OMEdit/OMEditGUI/CMakeLists.txt b/OMEdit/OMEditGUI/CMakeLists.txt
index bad8751c77..3e7d939de8 100644
--- a/OMEdit/OMEditGUI/CMakeLists.txt
+++ b/OMEdit/OMEditGUI/CMakeLists.txt
@@ -11,7 +11,7 @@ else()
   set(app_icon_macos "")
 endif()
 
-add_executable(OMEdit WIN32 MACOSX_BUNDLE main.cpp rc_omedit.rc ${app_icon_macos})
+add_executable(OMEdit WIN32 main.cpp rc_omedit.rc ${app_icon_macos})
 target_link_libraries(OMEdit PRIVATE OMEditLib)
 
 if(APPLE)
@@ -69,7 +69,6 @@ if(OM_OMEDIT_INSTALL_RUNTIME_DLLS AND MINGW)
             DIRECTORIES ${MSYSTEM_PREFIX_ESCAPED}/bin ${RUNTIME_LIB_DIRS}
             PRE_EXCLUDE_REGEXES "api-ms-" "ext-ms-"
             POST_EXCLUDE_REGEXES ".*system32/.*\\.dll")
-else()
-  install(TARGETS OMEdit
-          BUNDLE DESTINATION ${OM_MACOS_INSTALL_BUNDLEDIR})
+else ()
+  omc_install_gui_client(OMEdit)
 endif()
diff --git a/OMNotebook/OMNotebook/OMNotebookGUI/CMakeLists.txt b/OMNotebook/OMNotebook/OMNotebookGUI/CMakeLists.txt
index d9e6102869..0a3a816cee 100644
--- a/OMNotebook/OMNotebook/OMNotebookGUI/CMakeLists.txt
+++ b/OMNotebook/OMNotebook/OMNotebookGUI/CMakeLists.txt
@@ -100,7 +100,7 @@ else()
   set(app_icon_macos "")
 endif()
 
-add_executable(OMNotebook WIN32 MACOSX_BUNDLE ${OMNOTEBOOKLIB_SOURCES} ${OMNOTEBOOKLIB_HEADERS} rc_omnotebook.rc ${app_icon_macos})
+add_executable(OMNotebook WIN32 ${OMNOTEBOOKLIB_SOURCES} ${OMNOTEBOOKLIB_HEADERS} rc_omnotebook.rc ${app_icon_macos})
 target_compile_definitions(OMNotebook PRIVATE OMNOTEBOOKLIB_MOC_INCLUDE)
 
 target_include_directories(OMNotebook PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
@@ -116,13 +116,12 @@ endif ()
 target_link_libraries(OMNotebook PUBLIC OMPlotLib)
 target_link_libraries(OMNotebook PUBLIC OpenModelicaCompiler)
 
+omc_install_gui_client(OMNotebook)
 if(APPLE)
   set_target_properties(OMNotebook PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${CMAKE_CURRENT_SOURCE_DIR}/Info.plist)
 endif()
 
 
-install(TARGETS OMNotebook
-        BUNDLE DESTINATION ${OM_MACOS_INSTALL_BUNDLEDIR})
 install(FILES stylesheet.xml
               commands.xml
         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/omnotebook/)
diff --git a/OMPlot/OMPlot/OMPlotGUI/CMakeLists.txt b/OMPlot/OMPlot/OMPlotGUI/CMakeLists.txt
index a1d187d953..86e25782dd 100644
--- a/OMPlot/OMPlot/OMPlotGUI/CMakeLists.txt
+++ b/OMPlot/OMPlot/OMPlotGUI/CMakeLists.txt
@@ -64,10 +64,8 @@ else()
   set(app_icon_macos "")
 endif()
 
-add_executable(OMPlot WIN32 MACOSX_BUNDLE main.cpp rc_omplot.rc ${app_icon_macos})
+add_executable(OMPlot WIN32 main.cpp rc_omplot.rc ${app_icon_macos})
 target_link_libraries(OMPlot PRIVATE OMPlotLib)
 
-
 install(TARGETS OMPlotLib)
-install(TARGETS OMPlot
-        BUNDLE DESTINATION ${OM_MACOS_INSTALL_BUNDLEDIR})
+omc_install_gui_client(OMPlot)
diff --git a/OMShell/OMShell/OMShellGUI/CMakeLists.txt b/OMShell/OMShell/OMShellGUI/CMakeLists.txt
index ed5598ddf2..6959172bc4 100644
--- a/OMShell/OMShell/OMShellGUI/CMakeLists.txt
+++ b/OMShell/OMShell/OMShellGUI/CMakeLists.txt
@@ -27,17 +27,13 @@ else()
   set(app_icon_macos "")
 endif()
 
-add_executable(OMShell WIN32 MACOSX_BUNDLE ${OMSHELLL_SOURCES}
-                                           ${OMSHELLL_HEADERS}
-                                           ${app_icon_macos})
+add_executable(OMShell WIN32 ${OMSHELLL_SOURCES} ${OMSHELLL_HEADERS} ${app_icon_macos})
 
 target_link_libraries(OMShell PRIVATE OpenModelicaCompiler)
 target_link_libraries(OMShell PRIVATE Qt${OM_QT_MAJOR_VERSION}::Xml)
 target_link_libraries(OMShell PRIVATE Qt${OM_QT_MAJOR_VERSION}::Widgets)
 target_link_libraries(OMShell PRIVATE Qt${OM_QT_MAJOR_VERSION}::PrintSupport)
 
-
-install(TARGETS OMShell
-        BUNDLE DESTINATION ${OM_MACOS_INSTALL_BUNDLEDIR})
+omc_install_gui_client(OMShell)
 install(FILES commands.xml
         DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/omshell)
diff --git a/cmake/omc_utils.cmake b/cmake/omc_utils.cmake
index b64137ae2b..b22978a4bd 100644
--- a/cmake/omc_utils.cmake
+++ b/cmake/omc_utils.cmake
@@ -21,3 +21,15 @@ macro(omc_option var help_text value)
   option(${var} ${help_text} ${value})
   omc_add_to_report(${var})
 endmacro(omc_option)
+
+macro(omc_install_gui_client target)
+  option(OM_MACOS_APP_BUNDLE "Build gui clients as application bundles on macos" ON)
+  mark_as_advanced(OM_MACOS_APP_BUNDLE)
+  # On macOS we want BUNDLEs (.app) to go to an 'Applications/' directory instead of a 'bin/' directory
+  if(APPLE AND OM_MACOS_APP_BUNDLE)
+    set_target_properties(${target} PROPERTIES MACOSX_BUNDLE TRUE)
+  endif ()
+  set(OM_MACOS_INSTALL_BUNDLEDIR "Applications")
+  install(TARGETS ${target} RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+                            BUNDLE DESTINATION ${OM_MACOS_INSTALL_BUNDLEDIR})
+endmacro()
