diff --git a/fmi4c/src/fmi4c.c b/fmi4c/src/fmi4c.c
index c7e2670..22b1856 100644
--- a/OMSimulator/3rdParty/fmi4c/src/fmi4c.c
+++ b/OMSimulator/3rdParty/fmi4c/src/fmi4c.c
@@ -22,14 +22,20 @@
 #include <sys/stat.h>
 #endif
 
-#if defined(__x86_64__) || defined(_M_X64)
-    #define arch_str "x86_64"
+#if defined(__x86_64__) || defined(_M_X64) || defined(__aarch64__) || defined(__arm64__) || defined(_M_ARM64)
     #define bits_str "64"
 #else
-    #define arch_str "x86"
     #define bits_str "32"
 #endif
 
+#if defined(__aarch64__) || defined(__arm64__) || defined(_M_ARM64)
+    #define arch_str "aarch64"
+#elif defined(__x86_64__) || defined(_M_X64)
+    #define arch_str "x86_64"
+#else
+    #define arch_str "x86"
+#endif
+
 #if defined(__CYGWIN__)
     #define dirsep_str "/"
 #elif defined(_WIN32)
@@ -42,6 +48,10 @@
     #define dllext_str ".dll"
     #define fmi12_system_str "win"
     #define fmi3_system_str "windows"
+#elif defined(__APPLE__)
+    #define dllext_str ".dylib"
+    #define fmi12_system_str "darwin"
+    #define fmi3_system_str "darwin"
 #else
     #define dllext_str ".so"
     #define fmi12_system_str "linux"
