diff --git a/OMCompiler/Compiler/FrontEndCpp/CMakeLists.txt b/OMCompiler/Compiler/FrontEndCpp/CMakeLists.txt
index 0379f57a40..bb36fc8ae7 100644
--- a/OMCompiler/Compiler/FrontEndCpp/CMakeLists.txt
+++ b/OMCompiler/Compiler/FrontEndCpp/CMakeLists.txt
@@ -41,8 +41,8 @@ target_link_libraries(omcfrontendcpp PUBLIC omc::simrt::runtime)
 
 target_include_directories(omcfrontendcpp PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
-set(CC ${CMAKE_C_COMPILER})
-set(CXX ${CMAKE_CXX_COMPILER})
+get_filename_component(CC ${CMAKE_C_COMPILER} NAME)
+get_filename_component(CXX ${CMAKE_CXX_COMPILER} NAME)
 
 if (MSVC)
 
diff --git a/OMCompiler/Compiler/runtime/CMakeLists.txt b/OMCompiler/Compiler/runtime/CMakeLists.txt
index 285de89e8d..5713b7186e 100644
--- a/OMCompiler/Compiler/runtime/CMakeLists.txt
+++ b/OMCompiler/Compiler/runtime/CMakeLists.txt
@@ -184,9 +184,9 @@ endif()
 
 set(host_short ${CMAKE_LIBRARY_ARCHITECTURE})
 
-set(RUNTIMECC ${CMAKE_C_COMPILER})
-set(CC ${CMAKE_C_COMPILER})
-set(CXX ${CMAKE_CXX_COMPILER})
+get_filename_component(RUNTIMECC ${CMAKE_C_COMPILER} NAME)
+get_filename_component(CC ${CMAKE_C_COMPILER} NAME)
+get_filename_component(CXX ${CMAKE_CXX_COMPILER} NAME)
 
 find_package(OpenMP)
 if(OpenMP_FOUND)
